src/App.js

```diff
@@ -45,6 +45,14 @@ const App = () => {
         const unregister = chrome.on('GLOBAL_FILTER_UPDATE', ({ data }) => {
             const [workloads, SIDs, tags] = chrome.mapGlobalFilter(data, true, true) ?? [null, null, null];
             dispatch(setGlobalFilter({ workloads, SIDs, tags }));
+            // Always the global filter to the URL
+            const url = new URL(window.location);
+            const hashSearchParams = new URLSearchParams();
+            hashSearchParams.set('workloads', Object.keys(workloads).toString());
+            hashSearchParams.set('SIDs', SIDs.toString());
+            hashSearchParams.set('tags', tags.toString());
+            url.hash = hashSearchParams.toString();
+            window.history.replaceState(null, null, url);
```

Comment:
As there are already react-router [useSearchParams](https://reactrouter.com/en/main/hooks/use-search-params) hooks available, would it be better to use that instead of manipulating window history object?

Reply:
@bastilian should also be working on URL params for the OS filter. Maybe we can sync up and agree on the best approach.

Reply:
My changes shouldn't conflict with this. But yes, we should be using the `useSearchParams` hook here if possible.

Reply:
I was afraid it would cause a rerender so I didn't even bother, but looks like it works. I'll be using `navigate("#workloads...")` since searchParams doesnt work with hashes in the url. Thanks for the reviews :smile: